src/Microsoft.AspNetCore.JsonPatch/Helpers/ExpressionHelpers.cs

```diff
@@ -93,6 +70,26 @@ private static string GetPath(Expression expr, bool firstTime)
             }
         }
 
+        private static string GetPropertyNameFromMemberExpression(MemberExpression memberExpression)
+        {
+            // if there's a JsonProperty attribute, we must return the PropertyName
+            // from the attribute rather than the member name 
+            var jsonPropertyAttribute =
+                memberExpression.Member.GetCustomAttributes(
+                typeof(JsonPropertyAttribute), true);
+
+            // to object[] to avoid LINQ
+            var jsonPropertyAttributeAsObjectArray = (object[])jsonPropertyAttribute;
```

Comment:
Aha, now I get it ;-) Couldn't avoid LINQ without the downcast, but I missed the method to get one attribute instead of having to go through getting the list & then the first one from the list.  Should be fixed now :)